iib/workers/tasks/build.py

```diff
@@ -991,14 +1004,14 @@ def handle_add_request(
         )
 
     set_request_state(request_id, 'in_progress', 'Creating the manifest list')
-    output_pull_spec = _create_and_push_manifest_list(request_id, arches)
+    output_pull_specs = _create_and_push_manifest_list(request_id, arches, build_tags)
     if legacy_support_packages:
         export_legacy_packages(
-            legacy_support_packages, request_id, output_pull_spec, cnr_token, organization
+            legacy_support_packages, request_id, output_pull_specs[0], cnr_token, organization
```

Comment:
could you please add a comment here as to why we are using the first element of the list?